code/game/turfs/floors/floor_icon.dm

```diff
@@ -7,14 +7,42 @@
 	/// A cache for trench images, keyed by icon and then by color.
 	VAR_PRIVATE/static/list/_trench_image_cache = list()
 
+/turf/floor/proc/can_draw_edge_over(turf/floor/turf_to_check)
+	if(istype(turf_to_check))
+		var/my_height    = get_physical_height()
+		var/their_height = turf_to_check.get_physical_height()
+		// Uppermost turfs draw over lower turfs if there is a serious difference.
+		if(my_height != their_height)
+			return my_height > their_height
+		// Use edge layer if we're within height range.
+		return can_layer_over(turf_to_check)
+	return TRUE
+
+/turf/floor/proc/can_layer_over(turf/floor/turf_to_check)
+	if(!istype(turf_to_check))
+		return FALSE
+	var/decl/flooring/my_flooring = get_topmost_flooring()
+	if(!istype(my_flooring) || my_flooring.icon_edge_layer == FLOOR_EDGE_NONE)
+		return FALSE
+	var/decl/flooring/their_flooring = turf_to_check.get_topmost_flooring()
+	if(!istype(their_flooring))
+		return TRUE
+	if(their_flooring?.type == my_flooring.neighbour_type)
```

Comment:
i guess this is fine since path material presets aren't subtypes now? so it'd just be `/decl/flooring/path/herringbone` instead of `/turf/floor/natural/path/herringbone/basalt`